sw/device/tests/sram_ctrl_execution_test_main.c

```diff
@@ -0,0 +1,180 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/dif/dif_sram_ctrl.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/lc_ctrl_testutils.h"
+#include "sw/device/lib/testing/otp_ctrl_testutils.h"
+#include "sw/device/lib/testing/sram_ctrl_testutils.h"
+#include "sw/device/lib/testing/test_framework/test_main.h"
+#include "sw/device/lib/testing/test_framework/test_status.h"
+
+const test_config_t kTestConfig;
+
+static volatile const uint32_t kExecParam1 = 2;
+static volatile const uint32_t kExecParam2 = 3;
+
+static const uint32_t kRamStartBytes =
+    TOP_EARLGREY_SRAM_CTRL_MAIN_RAM_BASE_ADDR;
+static const uint32_t kRamEndBytes =
+    TOP_EARLGREY_SRAM_CTRL_MAIN_RAM_BASE_ADDR +
+    TOP_EARLGREY_SRAM_CTRL_MAIN_RAM_SIZE_BYTES - 1;
+
+/*
+ * OTP HW partition relative IFETCH offset in bytes.
+ *
+ * x = OTP_CTRL_PARAM_EN_SRAM_IFETCH_OFFSET (1728)
+ * y = OTP_CTRL_PARAM_HW_CFG_OFFSET (1664)
+ * IFETCH_OFFSET = (x - y) / 8 = 8
+ */
+static volatile const uint32_t kOtpIfetchHwRelativeOffset = 8;
```

Comment:
I don't think this needs to be volatile.

Also we should just do the calculation in the code rather than hardcoding 8.